modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/indexreader/IgniteIndexReader.java

```diff
@@ -1219,6 +1241,44 @@ private TreeNode getTreeNode(long pageId, TreeTraverseContext nodeCtx) {
         }
     }
 
+    /**
+     * @param nodeCtx tree traverse context.
+     * @param pageContent inner page content.
+     * @param io inner page io.
+     * @throws IgniteCheckedException
+     */
+    private void checkTriangleInvariant(TreeTraverseContext nodeCtx, PageContent pageContent, PageIO io)
+        throws IgniteCheckedException {
+        if (io instanceof BPlusInnerIO && pageContent.linkedPageIds != null) {
+            long rightChildPageId = -1;
+
+            final ByteBuffer innerBuf = allocateBuffer(pageSize);
+
+            try {
+                // linkedPageIds are populated in InnerPageIOProcessor.getContent
+                for (Long linkedPageId : pageContent.linkedPageIds) {
+                    triangleChecksNum++;
+
+                    innerBuf.rewind();
+
+                    if (rightChildPageId != -1 && linkedPageId != rightChildPageId)
+                        triangleErrorsCounter++;
```

Comment:
Would it make sense to preserve these `pageId`s and print them, like we do with orphans?

Reply:
fixed